src/Uno.UI/UI/Xaml/Shapes/Shape.iOSmacOS.cs

```diff
@@ -0,0 +1,638 @@
+					//    ------
+					//    |\   |
+					//    | \  |
+					//    |  \ |
+					//    |   \|
+					//    ------
+					//
+					// So for measure when None we includes that origin in the path size.
+					//
+					// Also, as the path does not have any notion of stroke thickness, we have to include it for the measure phase.
+					// Note: The logic would say to include the full StrokeThickness as it will "overflow" half on booth side of the path,
+					//		 but WinUI does include only the half of it.
+					var pathNaturalSize = new Size(
+						pathBounds.X == 0 ? pathBounds.Width + strokeThickness : pathBounds.Right + halfStrokeThickness,
+						pathBounds.Y == 0 ? pathBounds.Height + strokeThickness : pathBounds.Bottom + halfStrokeThickness);
+					size = pathNaturalSize.AtMost(userMaxSize).AtLeast(userMinSize); // The size defined on the Shape has priority over the size of the geometry itself!
+					break;
+
+				case Stretch.Fill:
+					size = userMaxSize.FiniteOrDefault(availableSize);
+					break;
+
+				case Stretch.Uniform:
+					size = userMaxSize.FiniteOrDefault(availableSize);
+					renderScale = ComputeScaleFactors(pathSize, ref size, strokeThickness);
+					if (renderScale.x > renderScale.y)
+					{
+						renderScale.x = renderScale.y;
+						size.Width = pathSize.Width * renderScale.x + strokeThickness;
+					}
+					else
+					{
+						renderScale.y = renderScale.x;
+						size.Height = pathSize.Height * renderScale.y + strokeThickness;
+					}
+					break;
+
+				case Stretch.UniformToFill:
+					size = userMinSize.AtLeast(availableSize);
+					renderScale = ComputeScaleFactors(pathSize, ref size, strokeThickness);
+					if (renderScale.x < renderScale.y)
+					{
+						renderScale.x = renderScale.y;
+						size.Width = pathSize.Width * renderScale.x + strokeThickness;
+					}
+					else
+					{
+						renderScale.y = renderScale.x;
+						size.Height = pathSize.Height * renderScale.y + strokeThickness;
+					}
+					break;
+			}
+
+			return size;
+		}
+
+		private protected Size ArrangeAbsoluteShape(Size finalSize, CGPath path)
+		{
+			if (path == null)
+			{
+				Render(null);
+				return default;
+			}
+
+			var stretch = Stretch;
+			var (userMinSize, userMaxSize) = this.GetMinMax();
+			var userSize = GetUserSize();
+			var strokeThickness = StrokeThickness;
+			var halfStrokeThickness = GetHalfStrokeThickness();
+			var pathBounds = path.BoundingBox;
+			var pathSize = (Size)pathBounds.Size;
+
+			if (nfloat.IsInfinity(pathBounds.Right) || nfloat.IsInfinity(pathBounds.Bottom))
+			{
+				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+				{
+					this.Log().Debug($"Ignoring path with invalid bounds {pathBounds}");
+				}
+
+				return default;
+			}
+
+			// Compute the final size of the Shape and the render properties
+			Size size;
+			(double x, double y) renderOrigin, renderScale, renderOverflow;
+			switch (stretch)
+			{
+				default:
+				case Stretch.None:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty 'case' clause.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5363624)